nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -67,7 +69,13 @@ private boolean exists(URI publicationId) {
 
     private void createCandidate(CandidateEvaluatedMessage evaluatedCandidate) {
         var pendingCandidate = toPendingCandidate(evaluatedCandidate.candidateDetails());
-        nviCandidateRepository.save(pendingCandidate);
+        var candidateDb = pendingCandidate.toDb();
+        nviCandidateRepository.save(candidateDb);
+    }
+
+    @JacocoGenerated //TODO: Remove when its actually used
+    public Optional<Candidate> findByPublicationId(URI publicationId) {
```

Comment:
Since we are in nviService, we need to stop using publication related things, since publicationId and candidateId are the same, maybe use candidateId instead. Rename method to findByCandidateId

Reply:
Maybe return Candidate and not Optional<Candidate>